### DISCUSSION TITLE
Dynamic fields

### DISCUSSION THREAD

**Discussion by oscarjb1** [react-querybuilder/react-querybuilder#817] [Q&A]
Hello community,

I have a case where I need to perform searches on unstructured documents, so it is not possible to have a list of fields, instead, I would expect the user to be able to freely type the field name, like in a TextField instead of a Select. Is this possible?

**Comment by oscarjb1** [comment]
I found similar thread: https://github.com/react-querybuilder/react-querybuilder/discussions/347

**Reply by jakeboone02** [reply]
Yeah, I was going to answer basically the same as https://github.com/react-querybuilder/react-querybuilder/discussions/347#discussioncomment-3224278. Is there anything else you need, or will that cover the requirement?

**Reply by oscarjb1** [reply]
Yes, one more thing, although I want to display a TextField, I also have preconfigured fields, so what I want to do is display a Select or an Input Text depending on the field, however, I don't know how to pass metadata to the Fields, so that fieldSelector can know whether to display a Select or an Input, I initially thought of using the props.rule.field field, however, in an Input Text, this changes as I write about the input, would it be possible to send it some metadata to distinguish it from the rest, something like the `my-input-field` field:

```
const fields = [
    { name: 'lastReceivedAt', label: 'Created At'},
    { name: 'my-input-field', placeholder: "Metadata", label: 'Metadata', metadata: {name: "myInputTextField"} } <<<<<<<<<<<<
];

```

This is my FieldSelector:

```
const FieldSelector = (props) => {
    if (props.rule.field === "my-input-field") {
        return (
            <TextField variant="outlined" size="small"
                value={props.value}
                onChange={(e) => props.handleOnChange(e.target.value)}
                className={props.className}
            />
        )
    } else {
        return <MaterialValueSelector {...props} variant="outlined" size="small" />
    }
}
```

**Reply by jakeboone02** [reply]
I'm not sure what you're going for there, because the `field` property _is_ what you're updating with a field selector. As soon as you run `props.handleOnChange(e.target.value)` from a text input, the `props.rule.field` value will no longer be `"my-input-field"` and your selector will revert to `MaterialValueSelector`.
